### ISSUE TITLE
Grouping for array dimension causing error

### ISSUE THREAD

**Issue by namannehra** [crossfilter/crossfilter#155]
`crossfilter2` version: 1.5.2

Code:
```js
'use strict'

const crossfilter = require('crossfilter2')

const filter = crossfilter()
const tagsDimension = filter.dimension(e => e.tags, true)
const tagsGroup = tagsDimension.group()

const data1 = [
    {
        id: '0',
        tags: ['LaunchedBy', 'Name'],
    },
    {
        id: '1',
        tags: ['environment'],
    }
]
filter.add(data1)

filter.remove(e => e.id === '0')

const data2 = [
    {
        id: '0',
        tags: ['LaunchedBy', 'Name'],
    }
]
filter.add(data2)

const tagGroupings = tagsGroup.top(Infinity)
console.log('tag groupings', tagGroupings)
```

Error: 
```
TypeError: Cannot read property 'value' of undefined
    at resetMany (crossfilter-test/node_modules/crossfilter2/crossfilter.js:1566:39)
    at all (crossfilter-test/node_modules/crossfilter2/crossfilter.js:1617:28)
    at Object.top (crossfilter-test/node_modules/crossfilter2/crossfilter.js:1623:28)
    at Object.<anonymous> (crossfilter-test/index.js:31:32)
```

**Comment by namannehra** [comment]
`undefined` being assigned to `groupIndex[i0][index1]` at https://github.com/crossfilter/crossfilter/blob/master/src/index.js#L943 maybe the cause of this.

**Comment by gordonwoodhull** [comment]
Thanks for the report and test case, @namannehra! That code certainly looks like it should work fine.

We recently closed a similar issue #127 having to do with adding records while a tag dimension filter was active. Tag / array dimensions are a relatively new feature and all the bugs have not been worked out, see also #141.

**Comment by gordonwoodhull** [comment]
Fixed by #156 and released in 1.5.3
